letsencrypt/renewer.py

```diff
@@ -0,0 +1,117 @@
+"""Renewer tool to handle autorenewal and autodeployment of renewed
+certs within lineages of successor certificates, according to
+configuration."""
+
+# TODO: sanity checking consistency, validity, freshness?
+
+# TODO: call new installer API to restart servers after deployment
+
+# TODO: when renewing or deploying, update config file to
+#       memorialize the fact that it happened
+
+import configobj
+import os
+
+from letsencrypt import configuration
+from letsencrypt import client
+from letsencrypt import crypto_util
+from letsencrypt import notify
+from letsencrypt import storage
+from letsencrypt.plugins import disco as plugins_disco
+
+DEFAULTS = configobj.ConfigObj("renewal.conf")
+DEFAULTS["renewal_configs_dir"] = "/tmp/etc/letsencrypt/configs"
+DEFAULTS["official_archive_dir"] = "/tmp/etc/letsencrypt/archive"
+DEFAULTS["live_dir"] = "/tmp/etc/letsencrypt/live"
+
+class AttrDict(dict):
+    """A trick to allow accessing dictionary keys as object
+    attributes."""
+    def __init__(self, *args, **kwargs):
+        super(AttrDict, self).__init__(*args, **kwargs)
+        self.__dict__ = self
+
+def renew(cert, old_version):
+    """Perform automated renewal of the referenced cert, if possible."""
```

Comment:
missing `:param:` (it's e.g. not clear to me why `cert.configfile` would be available...)